### PR TITLE
Make example string conform to regex

### PR DESCRIPTION
**Pull request by dewijones92**
### Description
Make the example string conform to the regex pattern

### Motivation and Context
Otherwise the request body is invalid

### How Has This Been Tested?
Created unit test for it

### My PR contains... 

- [ ] No code changes (`src/` is unmodified: changes to documentation, CI, metadata, etc.)
- [x] Dependency changes (any modification to dependencies in `package.json`)
- [ ] Bug fixes (non-breaking change which fixes an issue)
- [x] Improvements (misc. changes to existing features)
- [x] Features (non-breaking change which adds functionality)

### My changes...
- [ ] are breaking changes to a public API (config options, System API, major UI change, etc).
- [ ] are breaking changes to a private API (Redux, component props, utility functions, etc.).
- [ ] are breaking changes to a developer API (npm script behavior changes, new dev system dependencies, etc).
- [x] are not breaking changes.

### Documentation
- [x] My changes do not require a change to the project documentation.
- [ ] My changes require a change to the project documentation.
- [ ] If yes to above: I have updated the documentation accordingly.

### Automated tests
- [ ] My changes can not or do not need to be tested.
- [x] My changes can and should be tested by unit and/or integration tests.
- [x] If yes to above: I have added tests to cover my changes.
- [x] If yes to above: I have taken care to cover edge cases in my tests.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by dewijones92** [comment]
I have cleaned up the code

**Comment by dewijones92** [comment]
Rebased

**Comment by dewijones92** [comment]
Any particular reason why this isn't being merged in? Am I missing something?
thanks :)

**Comment by tim-lai** [comment]
@dewijones92 Thanks for the PR, it functionally works for me. However, can you update so that it passes the `lint-errors` test? 

```
9:42  error  'generateStringFromRegex' was used before it was defined  no-use-before-define
```

File: src/core/plugins/samples/fn.js [hunk 1]
```diff
@@ -3,9 +3,10 @@ import isEmpty from "lodash/isEmpty"
 import { objectify, isFunc, normalizeArray, deeplyStripKey } from "core/utils"
 
 import memoizeN from "../../../helpers/memoizeN"
+import RandExp from "randexp"
```
Comment (tim-lai) [inline src/core/plugins/samples/fn.js:6]: This import should be grouped with lines 1-2

File: package.json [hunk 2]
```diff
@@ -74,6 +74,7 @@
     "js-yaml": "=4.1.0",
     "lodash": "^4.17.21",
     "prop-types": "^15.7.2",
+    "randexp": "=0.5.3",
```
Comment (tim-lai) [inline package.json:77]: let's use a caret (^) here.

**Comment by dewijones92** [comment]
rebased

**Comment by tim-lai** [comment]
@dewijones92 PR merged! Thanks for the contribution!

**Comment by GandalfTheBlack16** [comment]
Hi there, I just opened an issue relative to this feature.., it's look like Swagger-ui now expects a @Schema Example attribute to be always specified, cause if you define a @Pattern on your field but you dont specify an @Schema(example="") value something go wrong and try it out section is so verbose by 'error': 

image


You can check the issue here: https://github.com/swagger-api/swagger-ui/issues/8003

And the repo with a easily reproducible demo here: https://github.com/GandalfTheBlack16/demo-swagger

Thanks so much!

**Comment by tautvydasfrontit** [comment]
In our project API models contains lots of regexp pattern such as this: "[\\p{L}]+(['՚’＇֊᠆‐‑﹣－-]+[\\p{L}]+)*[.]?([\\p{Zs}]+[\\p{L}]+(['՚\’＇֊᠆‐‑﹣－-]+[\\p{L}]+)*[.]?)*".
Since now swagger-ui tries to generate value for each field marked with 'pattern', it causes slow UI behaviour and sometimes even crash browser. Both cases swagger-ui becomes unusable.
I would appreciate a way to disable this functionality or have a workaround, cause now it really troubles our workflow.

**Pull request merged**